calico_kubernetes/calico_kubernetes.py

```diff
@@ -97,6 +147,223 @@ def _configure_interface(self):
         print('Finished configuring network interface')
         return ep
 
+    def _configure_profile(self, endpoint):
+        """
+        Configure the calico profile for a pod.
+
+        Currently assumes one pod with each name.
+        """
+        print('Configuring Pod Profile')
+        profile_name = self.pod_name
+
+        if self._datastore_client.profile_exists(profile_name):
+            print("A profile with name %s already exists" % profile_name)
+            sys.exit(1)
+        else:
+            self._datastore_client.create_profile(profile_name)
+
+        pod = self._get_pod_config()
+
+        self._apply_rules(profile_name)
+
+        self._apply_tags(profile_name, pod)
+
+        # Also set the profile for the workload.
+        print('Setting profile %s on endpoint %s' %
+              (profile_name, endpoint.endpoint_id))
+        self._datastore_client.set_profiles_on_endpoint(
+            profile_name, endpoint_id=endpoint.endpoint_id
+        )
+        print('Finished configuring profile.')
+
+    def _container_add(self, container_name, ip, interface):
+        """
+        Add a container (on this host) to Calico networking with the given IP.
+
+        :param container_name: The name or ID of the container.
+        :param ip: An IPAddress object with the desired IP to assign.
+        :param interface: The name of the interface in the container.
+        """
+        # Obtain information from Docker Client and validate container state
+        info = self._get_container_info(container_name)
```

Comment:
Can we move the `get_container_info` call out of `_container_add`, and instead add parameters to the function for `container_id` and `pid`? For mesos, those parameters aren't available through the docker client